apps/explorer/src/hooks/useGetCoins.ts

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useRpcClient } from '@mysten/core';
+import { useInfiniteQuery } from '@tanstack/react-query';
+
+import type { SuiAddress } from '@mysten/sui.js';
+
+const MAX_COINS_PER_REQUEST = 10;
+
+// Fetch all coins for an address, this will keep calling the API until all coins are fetched
+export function useGetCoins(
+    coinType: string,
+    address?: SuiAddress | null,
+    cacheTime = 0
```

Comment:
Probably should be null or undefined instead of 0 so that it falls back into the default behavior?